src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -285,5 +284,9 @@ public override ImmutableArray<CSharpAttributeData> GetAttributes()
         public bool HasEnumeratorCancellationAttribute => _baseParameterForAttributes?.HasEnumeratorCancellationAttribute ?? false;
 
         internal override MarshalPseudoCustomAttributeData? MarshallingInformation => _baseParameterForAttributes?.MarshallingInformation;
+
+        internal override bool IsMetadataOptional => _baseParameterForAttributes?.IsMetadataOptional == true;
+
+        internal override ConstantValue? ExplicitDefaultConstantValue => _baseParameterForAttributes?.ExplicitDefaultConstantValue;
```

Comment:
I'm not sure if there are other properties I should override, and what they'll affect.

Reply:
I looked over the `ParameterSymbol` members and all that comes to mind is `IsIDispatchConstant` and `IsIUnknownConstant`.

I think that synthesized parameter symbols with non-null `_baseParameterForAttributes` are only produced for lowering/emit, so we don't expect to actually check the `IsIDispatchConstant` or `IsIUnknownConstant` members from the `_baseParameterForAttributes`.

I am not certain whether it makes more sense to:
1. Do nothing to override or verify `IsIDispatchConstant` or `IsIUnknownConstant` on `SynthesizedComplexParameterSymbol` (i.e. don't change the implementation in this PR)
2. Change `IsIDispatchConstant` and `IsIUnknownConstant` on `SynthesizedParameterSymbolBase` to `throw ExceptionUtilities.Unreachable`, since we don't expect to check those members on such parameters, because we don't expect to bind a default argument based on such a parameter, for example. (I'm not certain about this--if it results in new CI failures then we'll have learned something about the usage :smile:)
3. Override `IsIDispatchConstant` and `IsIUnknownConstant` on `SynthesizedComplexParameterSymbol` to delegate to the `_baseParameterForAttributes`.

I'm leaning slightly toward (2).

Reply:
@RikkiGibson Thanks. I took the second approach as you suggested.

Is there anything needed for nullable analysis here?

e.g, override `FlowAnalysisAnnotation` and things related to nullable analysis?

Reply:
I had a few more overrides and assertions in mind but I felt like enumerating them was equal work to adding the implementations, so I went ahead and pushed a commit to your PR.